src/main/java/uk/gov/hmcts/probate/service/CaveatQueryService.java

```diff
@@ -106,14 +96,21 @@ public CaveatData findCaveatById(CaseType caseType, String caveatId) {
 
         ReturnedCaveats returnedCaveats;
         try {
-            returnedCaveats = restTemplate.postForObject(uri, entity, ReturnedCaveats.class);
+            returnedCaveats = nonNull(restTemplate.postForObject(uri, entity, ReturnedCaveats.class));
         } catch (HttpClientErrorException e) {
             appInsights.trackEvent(REST_CLIENT_EXCEPTION, e.getMessage());
             throw new CaseMatchingException(e.getStatusCode(), e.getMessage());
+        }catch (IllegalStateException e) {
+            throw new ClientDataException(e.getMessage());
         }
 
-        appInsights.trackEvent(REQUEST_SENT, uri.toString());
 
+        appInsights.trackEvent(REQUEST_SENT, uri.toString());
         return returnedCaveats.getCaveats();
     }
+
+    private static <T> T nonNull(@Nullable T result) {
```

Comment:
do we really need @Nullable here?

Reply:
its just to indicate that the value passed could have null values , if you suggest to get rid of it can try that out as well